src/pykiso/lib/connectors/trc_handler.py

```diff
@@ -0,0 +1,322 @@
+##########################################################################
+# Copyright (c) 2010-2022 Robert Bosch GmbH
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+##########################################################################
+
+"""
+TRC file handler based on Python CAN
+************************************
+
+:module: trc_handler
+
+:synopsis: Additional layer for python-can trc reader and write to handle can FD
+
+.. currentmodule:: trc_handler
+
+"""
+import logging
+from typing import List, Optional
+from datetime import datetime, timedelta
+
+try:
+    from can.io.trc import TRCFileVersion, TRCReader, TRCWriter
+    from can.util import channel2int, dlc2len, len2dlc
+    from can import Message
+except ImportError as e:
+    raise ImportError(
+        f"{e.name} dependency missing, consider installing pykiso with 'pip install pykiso[can]'"
+    )
+
+
+log = logging.getLogger(__name__)
+
+
+class TypedMessage(Message):
+    """Message with type attribut added and that can handle empty arbitration id.
+    Necessary because python-can does not handle types from CAN FD and informations
+    contained in Message are not enough to deduce it.
+    Some of those types (ex: ST) also have no arbitration id which cannot be handle by
+    python-can Message object.
+    """
+
+    def __init__(self, type: str, msg: Message):
```

Comment:
I don't know the details of the Message class, but it seems strange to use an object of the parent class to initialize the child class. Maybe since you're using the attributes of the parent class you can pass them like creating an object of type Message, just including the type?

Reply:
Definitely, I switched to proper inheritance instead :)